File: php7/src/ChainOfResponsibility/Agent/TillAgent.php
<?php

declare(strict_types = 1);


namespace PetrePatrasc\ChainOfResponsibility\Agent;


use PetrePatrasc\ChainOfResponsibility\LoanApplicationRequest;
use PetrePatrasc\ChainOfResponsibility\LoanApplicationResponse;

class TillAgent extends AbstractLoanAgent
{
    /**
     * @inheritDoc
     */
    public function handleLoanAmount(LoanApplicationRequest $loanRequest): LoanApplicationResponse
    {
        if ($loanRequest->getAmount() < 5000) {
            return $this->approveLoanRequest();
        }

        return $this->getNextInChain()->handleLoanAmount($loanRequest);
    }
}

File: java/src/ro/petrepatrasc/chainofresponsability/agent/TillAgent.java
package ro.petrepatrasc.chainofresponsability.agent;

import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationRequest;
import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationResponse;

/**
 * @author Petre Pătrașc <petre@dreamlabs.ro>
 */
public class TillAgent extends AbstractLoanAgent {

    public static final float AGENT_THRESHOLD = 5000;

    @java.lang.Override
    public LoanApplicationResponse handleLoanRequest(LoanApplicationRequest loanRequest) {
        if (loanRequest.getAmount() < this.AGENT_THRESHOLD) {
            return this.approveLoanRequest();
        }

        return this.getNextInChain().handleLoanRequest(loanRequest);
    }
}
